nautobot_device_lifecycle_mgmt/filters.py

```diff
@@ -25,76 +26,91 @@
-    )
-
-    documentation_url = django_filters.CharFilter(
-        lookup_expr="contains",
+        label="Device Type (model)",
     )
 
-    end_of_support = django_filters.DateFilter()
-    end_of_support__gte = django_filters.DateFilter(field_name="end_of_support", lookup_expr="gte")
-    end_of_support__lte = django_filters.DateFilter(field_name="end_of_support", lookup_expr="lte")
-
-    end_of_sale = django_filters.DateFilter()
-    end_of_sale__gte = django_filters.DateFilter(field_name="end_of_sale", lookup_expr="gte")
-    end_of_sale__lte = django_filters.DateFilter(field_name="end_of_sale", lookup_expr="lte")
-
-    end_of_security_patches = django_filters.DateFilter()
-    end_of_security_patches__gte = django_filters.DateFilter(field_name="end_of_security_patches", lookup_expr="gte")
-    end_of_security_patches__lte = django_filters.DateFilter(field_name="end_of_security_patches", lookup_expr="lte")
-
-    end_of_sw_releases = django_filters.DateFilter()
-    end_of_sw_releases__gte = django_filters.DateFilter(field_name="end_of_sw_releases", lookup_expr="gte")
-    end_of_sw_releases__lte = django_filters.DateFilter(field_name="end_of_sw_releases", lookup_expr="lte")
-
-    expired = django_filters.BooleanFilter(method="expired_search", label="Expired")
-
     class Meta:
         """Meta attributes for filter."""
 
         model = HardwareLCM
 
         fields = [
+            "expired",
+            "device_type",
+            "inventory_item",
+            "release_date",
             "end_of_sale",
             "end_of_support",
             "end_of_sw_releases",
             "end_of_security_patches",
-            "inventory_item",
             "documentation_url",
-            "expired",
         ]
 
-    def search(self, queryset, name, value):  # pylint: disable=unused-argument, no-self-use
-        """Perform the filtered search."""
-        if not value.strip():
-            return queryset
-
-        qs_filter = Q(end_of_sale__icontains=value) | Q(end_of_support__icontains=value)
-        return queryset.filter(qs_filter)
-
-    def expired_search(self, queryset, name, value):  # pylint: disable=unused-argument, no-self-use
+    def _expired_search(self, queryset, name, value):  # pylint: disable=unused-argument, no-self-use
         """Perform the filtered search."""
         today = datetime.datetime.today().date()
-        lookup = "gte" if not value else "lt"
-
-        qs_filter = Q(**{f"end_of_sale__{lookup}": today}) | Q(**{f"end_of_support__{lookup}": today})
+        # End of support dates less than today are expired.
+        # End of support dates greater than or equal to today are not expired.
+        # If the end of support date is null, the notice will never be expired.
+        qs_filter = None
+        if value:
+            qs_filter = Q(**{"end_of_support__lt": today})
+        if not value:
+            qs_filter = Q(**{"end_of_support__gte": today}) | Q(**{"end_of_support__isnull": not value})
```

Comment:
I think having `not value` here is confusing, let's just do `True` instead.

```suggestion
            qs_filter = Q(**{"end_of_support__gte": today}) | Q(**{"end_of_support__isnull": True})
```

Reply:
Agreed, changed this to `True`